### PR TITLE
Update to tell if the year is a leap year and how many days in the year

### PR DESCRIPTION
**Pull request by ssherriff**
Add a bool method to check if is leap year, then also get days in the year based on whether it is a leap year or not.

### CODE REVIEW

File: date.go [hunk 1]
```diff
@@ -182,6 +182,21 @@ func (d Date) YearDay() int {
 	return d.Time().YearDay()
 }
 
+// LeapYear gives whether or not the year is a leap year.
+// Any year divisible by 4 except century years
+// Any century year divisible by 400
+func (d Date) LeapYear() bool {
```
Comment (17twenty) [inline date.go]: Might be just as easy to use time.YearDay?

Comment (17twenty) [reply date.go]: Makes the below function redundant then too

Comment (ssherriff) [reply date.go]: That's just the current day of the year though, right?

**Review by e-pratt — APPROVED** [review]

**Review by e-pratt — APPROVED** [review]

File: date.go (hunk 1 above)
Comment (17twenty) [reply date.go]: If you create a time object on 31/12/XX and call YearDay it would give you the days in the year. Either 365 or 366, pretty neat 💪

Comment (ssherriff) [reply date.go]: yeah, that's a different way of doing it, thanks for your suggestion

**Review by ramshaw888 — APPROVED** [review]
Lgtm

**Pull request merged**